### PR TITLE
Add user setting to opt-out from data collection

### PR DESCRIPTION
**Pull request by GretaD**
Add user setting to opt-out from data collection

Fixes https://github.com/nextcloud/mail/issues/2559

### CODE REVIEW

File: src/components/AppSettingsMenu.vue [hunk 1]
```diff
@@ -4,6 +4,22 @@
 			{{ t('mail', 'Add mail account') }}
 		</router-link>
 
+		<p v-if="loadingOptOutSettings" class="app-settings">
+			<span class="icon-loading-small"></span>
+			<!-- better label? -->
+			{{ t('mail', 'Collect data..') }}
```
Comment (ChristophWurst) [inline src/components/AppSettingsMenu.vue]: My take: "Allow the app to collection data about your interactions. Based on this data, the app will adapt to your preferences. The data will only be stored locally."

**Comment by GretaD** [comment]
> Lint isn't happy :)

it never is #spoiledLittleSomething

File: src/components/AppSettingsMenu.vue (hunk 1 above)
Comment (ChristophWurst) [reply src/components/AppSettingsMenu.vue]: you omitted the last two sentences on purpose? ;)

**Comment by ChristophWurst** [comment]
For the initial value on page load, see

* https://github.com/nextcloud/mail/blob/8b41503c90b9ab67626208dc25b7b786906173cd/lib/Controller/PageController.php#L128
* https://github.com/nextcloud/mail/blob/dfe651eee4d712cc3f252f9a2629143d0cff9fe8/templates/index.php#L34
* https://github.com/nextcloud/mail/blob/ad29c8a46cc0bc6e783e82cc81ea5c701c66fb98/src/main.js#L62-L65

File: src/components/AppSettingsMenu.vue (hunk 1 above)
Comment (GretaD) [reply src/components/AppSettingsMenu.vue]: haha nope, i thought you were explaining to me the first line.

Comment (ChristophWurst) [reply src/components/AppSettingsMenu.vue]: nah it's just a bit of text to clarify what "we" do with the data. so users know it's not sent somewhere suspicious but just the Nextcloud instance they're using

Comment (GretaD) [reply src/components/AppSettingsMenu.vue]: absolutely

File: lib/Controller/PageController.php [hunk 2]
```diff
@@ -126,6 +126,7 @@ public function index(): TemplateResponse {
 				'app-version' => $this->config->getAppValue('mail', 'installed_version'),
 				'accounts' => base64_encode(json_encode($accountsJson)),
 				'external-avatars' => $this->preferences->getPreference('external-avatars', 'true'),
+				'external-optOut' => $this->preferences->getPreference('external-optOut', 'true'),
```
Comment (ChristophWurst) [inline lib/Controller/PageController.php]: how about just `collect-data` as key?

Comment (GretaD) [reply lib/Controller/PageController.php]: sure, do you want to check it all and let me know if theres anything else i need to change?

**Comment by ChristophWurst** [comment]
I have to fix the tests

> 1) OCA\Mail\Tests\Unit\Controller\PageControllerTest::testIndex
>
>OCA\Mail\Contracts\IUserPreferences::getPreference('collect-data', 'true') was not expected to be called more than once.

**Pull request merged**